### ISSUE TITLE
Générer les altitudes + profiles de taxons

### ISSUE THREAD

**Issue by Splendens** [PnX-SI/gn_module_import#276]
Lorsqu'on n'a pas de champs "altitude" dans le fichier à importer, il est possible de générer les altitudes automatiquement lors de l'import. Cette option est sous forme d'une case à cocher, décochée par défaut. 
Cependant, les données sans altitude ne sont pas prises en compte pour les profils de taxons. 

Du coup, je me demandais s'il ne sera pas préférable de générer les altitudes par défaut si aucun champ "altitude" n'est mappé, sans même demander à l'utilisateur ?

**Comment by bouttier** [comment]
Générer des altitudes alors que la case est explicitement décoché me semble trompeur. Cependant, on peut l’activer par défaut.

À noter que l’on souhaiterai à priori renommer cette coche en « Générer les altitudes **manquantes** ». Cela veut dire que si les colonnes altitude_min / altitude_max ont été associées, alors les altitudes seront générées uniquement pour les lignes ne contenant pas d’altitude. Ainsi, activer cette coche par défaut me semble d’autant moins gênant, y compris pour importer des données dont l’altitude est fournie.

La valeur par défaut pourrait aussi être paramétrable, mais je suis pour limiter la prolifération à outrance des paramètres de configuration.

**Comment by DonovanMaillard** [comment]
Dans l'idée, je suis plutôt favorable aussi à ce que chaque transformation de la donnée soit le résultat d'un choix de l'utilisateur, et pas que le module effectue par défaut des transformations que l'utilisateur n'a pas choisi activement de faire. 

En complément en parallèle du fait que effectivement, la case à cocher permettra de calculer les altitudes manquantes et non plus toutes les altitudes, pour offrir toutes les possibilités aux utilisateurs qui importent leurs fichiers.

**Comment by camillemonchicourt** [comment]
On propose de simplifier et de toujours calculer les altitudes quand : 
- aucune altitude n'est mappée depuis le fichier source
- une colonne est mappée mais certaines lignes n'ont pas d'altitude renseignée dans le fichier source

Et idéalement de ne pas le faire au niveau du processus d'import, mais plus globalement au niveau du champs de la destination avec un trigger qui calcule automatiquement l'altitude si celle-ci n'est pas renseignée.
C'est déjà le fonctionnement en place au niveau des tables d'Occhab, Occtax et Monitoring.
A ajouter donc au niveau de la table de la Synthèse.

Ça évite d'avoir de la donnée hétérogène, avec parfois des altitudes et parfois non, avec les soucis en cascade que cela pose.
Et ça évite de poser des questions aux utilisateurs qu'ils ne comprennent pas tout le temps.
Ça simplifie aussi le processus et les calculs dans le module Import.

C'est pas vraiment de la transformation de données il me semble, mais du post-traitement de données absolues.

On aimerait faire pareil pour les UUID (pour ne plus avoir de données sans UUID dans GeoNature) mais c'est un autre sujet.

**Comment by camillemonchicourt** [comment]
Retour de @DonovanMaillard : pas idéal pour des données anciennes, biblio ou imprécise où on a de la données qu'à la commune par exemple.
Mais dommage pour autant d'avoir potentiellement pas mal de données précises sans altitude car on a laissé la possibilité à l'utilisateur de ne pas la calculer.

Donc OK pour lui pour supprimer cette possibilité et calculer l'altitude dans tous les cas où elle n'est pas fournie.

**Comment by DonovanMaillard** [comment]
Oui, ça permet surtout d'assurer la cohérence et l'homogénéité des informations sur un maximum de données, ça aide dans les calculs des profils etc, et ça simplifié les imports :)

**Comment by camillemonchicourt** [comment]
A voir, si on a uniquement alt_min ou alt_max de renseigné, alors on devrait le copier dans l'autre et non pas calculer que le manquant.
On devrait calculer que si alt_min ET alt_max ne sont pas fournis.

**Comment by DonovanMaillard** [comment]
C'est un cas un peu traitre oui... 

Si j'ai qu'un champs altitude je vais vouloir renseigner la même info dans les deux (mais c'est dans mon mapping!), si j'ai un champs min et max dans la source et que certaines sont vides il faut les calculer. 

Je dirais que si on a mappé correctement notre fichier, on est censé calculer tout ce qui est vide, pas aller chercher ce qui est dans l'autre champs. 

SI j'ai qu'un champs source -> je le mets dans les 2 champs cibles, donc je calcule là ou mon champs unique est vide. 

Si j'ai deux champs sources alt_min et alt_max, pas de raison d'aller récupérer l'autre valeur. 

Accessoirement c'est plus simple comme ca niveau développement et clarté : un champs altitude vide est calculé, point;

**Comment by camillemonchicourt** [comment]
Ah oui en effet, le cas que j'évoque devrait être géré au niveau du mapping.

Ce que tu proposes en certainement plus clair, homogène et cohérent.

**Comment by lpojgc** [comment]
ok avec la proposition de @DonovanMaillard mais avec un seul bémol, un peu alambiqué quand même : des données avec les 2 champs dans le fichier source mais un seul renseigné pour certaines lignes. Lors de l'import le calcul de l'autre pourrait ne pas être cohérent avec le premier (alt_min > alt_max ou inversement). A voir comment gérer dans ce cas-là, données écartées de l'import ou recalcul des 2 champs (ou ???).

**Comment by DonovanMaillard** [comment]
Bien vu @lpojgc il y a ce cas particulier. Mais finalement c'est comme quand tu as 2 champs source incohérents,  ta donnée est invalide et faut la traiter manuellement

Si j'ai une altitude min à 1800m sur une géométrie qui ne dépasse pas 900m c'est que le min est invalide. Mais j'imagine bien que ça peut être fréquent sur des fichiers de points ou l'altitude n'est pas exactement celle du mnt :/
